### ISSUE TITLE
[php-upload-progress] Importing crashes PHP CGI

### ISSUE THREAD

**Issue by jamboNum5** [phpmyadmin/phpmyadmin#16885]
I'm having issues importing a database on various browsers and OS's. Expected behaviour is that the database imports with no errors like it does on cli. sql file: https://pastebin.com/gU7MxRkY confirmed to work on the cli.

While using the sql import function on phpmyadmin with a very small db, I see an error in the browser "Could not load the progress of the import." This is curious as I have a very reasonable max execution time, max upload, memory allowance for php.

Apache error log shows, I tired upping the debug level but the error log doesn't seem to print much more than this.
```
[Fri May 07 08:23:20.661388 2021] [fcgid:warn] [pid 90931] (104)Connection reset by peer: [client 131.231.154.68:49954] mod_fcgid: error reading data from FastCGI server
[Fri May 07 08:23:20.661434 2021] [core:error] [pid 90931] [client 131.231.154.68:49954] End of script output before headers: import.php
```

In the main system log seems to show php-cgi crashing badly.
`[244762.305525] traps: php-cgi[90904] general protection ip:55e5e384cfcd sp:7ffe75e0c1f8 error:0 in php-cgi[55e5e358f000+44c000]` which seems to indicate the php-cgi has crashed.

I occasionally see a message in the browser saying : `Notice in ./import_status.php#102 Undefined index: Import_message`

I tried to use strace to work out at which point the problem was happening, but this was different when I tried importing a vanilla wordpress db.

```
lstat("/var/www/clients/client247", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/var/www/clients", {st_mode=S_IFDIR|0755, st_size=24576, ...}) = 0
lstat("/var/www", {st_mode=S_IFDIR|0755, st_size=45056, ...}) = 0
lstat("/var", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat("/var/www/clients/client247/web1839/web/vendor/composer/ClassLoader.php", {st_mode=S_IFREG|0644, st_size=14367, ...}) = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SI_KERNEL, si_addr=NULL} ---
+++ killed by SIGSEGV +++
```

- Operating system: CentOS 7.9.2009
- Web server: httpd-2.4.6-97.el7.centos.x86_64 / shared webhost ispconfig
- Database version: mariadb-server-5.5.68-1.el7.x86_64
- PHP version: remi php74-php-common-7.4.18-1.el7.remi.x86_64
- phpMyAdmin version: RELEASE_5_0_4 installed with git/composer/npm

I believe I have all the required packages for PHP to work correctly, if there is something glaringly obvious some one might be able to confirm here.

```
php74-php-pecl-uploadprogress-1.1.3-1.el7.remi.x86_64
php74-php-common-7.4.18-1.el7.remi.x86_64
php74-php-fpm-7.4.18-1.el7.remi.x86_64
php74-php-soap-7.4.18-1.el7.remi.x86_64
php74-php-bcmath-7.4.18-1.el7.remi.x86_64
php74-php-pear-1.10.12-5.el7.remi.noarch
php74-php-process-7.4.18-1.el7.remi.x86_64
php74-runtime-1.0-3.el7.remi.x86_64
php74-php-xml-7.4.18-1.el7.remi.x86_64
php74-php-xmlrpc-7.4.18-1.el7.remi.x86_64
php74-php-intl-7.4.18-1.el7.remi.x86_64
php74-php-gd-7.4.18-1.el7.remi.x86_64
php74-php-pecl-mcrypt-1.0.4-1.el7.remi.x86_64
php74-php-pecl-zip-1.19.2-1.el7.remi.x86_64
php74-php-mysqlnd-7.4.18-1.el7.remi.x86_64
php74-php-cli-7.4.18-1.el7.remi.x86_64
php74-php-json-7.4.18-1.el7.remi.x86_64
php74-php-opcache-7.4.18-1.el7.remi.x86_64
php74-php-pdo-7.4.18-1.el7.remi.x86_64
php74-php-mbstring-7.4.18-1.el7.remi.x86_64
```

**Comment by williamdes** [comment]
Hi!
I think this is the php bug: https://bugs.php.net/bug.php?id=80979

Could you try my test script to confirm?

**Comment by williamdes** [comment]
Also I have fixed the notice with https://github.com/phpmyadmin/phpmyadmin/commit/4d4acee8a01a0777e52b69413274993a9b02a631 and if you update to 5.1.0 or apply the patch it will disappear

**Comment by fyberoptik** [comment]
The update to RELEASE_5_1_0 has stopped the strace message, so thanks for helping with that one step forward.

Cheers for the second pointer, I get the same error with the test script you provided with php-cgi crashing. So I'm not alone with this problem, potentially not something I have done in error.
`[251105.528270] traps: php-cgi[102073] general protection ip:5620cf502fcd sp:7ffe9f2da678 error:0 in php-cgi[5620cf245000+44c000]` 

I'll keep an eye on that thread for some progress.

Many thanks for responding here @williamdes 👍

**Comment by fyberoptik** [comment]
I'm not sure if this is a particularly valid test, but I removed the `enctype="multipart/form-data"` element of the upload form and I still managed to trip the kernel which then killed php-cgi proccess. Not sure whether it is linked to the PHP bug you mentioned.

I'm struggling for ways to debug this given the limited information I can glean. This is a hunch, but it seems more likely that this is to do with using ispconfig and the config that support multiple versions of remi PHP.

**Comment by williamdes** [comment]
Hi @fyberoptik 
Thank you for the research work you did, if you could dig further that would be super helpfull
Maybe see https://github.com/php/pecl-php-uploadprogress/pull/9 as it could maybe fix the error

I have no knowledge on C code so I can not debug or fix this but If you have feel free to contribute

**Comment by fyberoptik** [comment]
Thanks for the pointer again I too, have no knowledge of C 😬. I just removed php74-php-pecl-uploadprogress and it has got around the problem and imported the database. It's obviously not a fix, but in this use case uploadprogress isn't really needed so it gives a work around at least.

In the link you provided, kid4git appears to be using a very similar combination of packages. I was using a combination of:

php74-php-pecl-uploadprogress-1.1.3-1.el7.remi.x86_64
php74-php-fpm-7.4.20-1.el7.remi.x86_64

**Comment by alabsi11** [comment]
Hi, I would like to know the structure of this [website](https://topdream.news/)

**Comment by fyberoptik** [comment]
This problem appears to be resolved via running with php8. I am now able to import sql files via phpmyadmin. Thanks for you help @williamdes.

**Comment by williamdes** [comment]
Seems to be fixed now, just tried on PHP `7.4.28` with `uploadprogress` `2.0.2`
